src/pgduckdb_types.cpp

```diff
@@ -1378,4 +1461,92 @@ InsertTupleIntoChunk(duckdb::DataChunk &output, duckdb::shared_ptr<PostgresScanG
 	scan_global_state->m_total_row_count++;
 }
 
+
+Numeric
+CreateNumeric(const NumericVar &var) {
```

Comment:
I think having this in its own file (preferably together with some other numeric code) would be nice. Or `pgduckdb_types.cpp` is getting quite big with a lot of random things in it. This function also seems pretty much entirely copied over from PG source code of the `make_result_opt_error` function. Would be nice to add that as a comment on the function.

Reply:
Discussed live:
* we'll vendor in `make_result_opt_error`
* we've checked that this function is called only directly from PG, never from a DuckDB thread